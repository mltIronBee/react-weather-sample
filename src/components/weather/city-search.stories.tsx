import React, { ComponentProps } from "react";
import { Story } from "@storybook/react/types-6-0";
import { CitySearch } from "@components/weather";

const Template: Story<ComponentProps<typeof CitySearch>> = (args) => <CitySearch {...args} />;

export const Default = Template.bind({});

Default.args = {
	value: "Odesa",
};

export const Loading = Template.bind({});

Loading.args = {
	value: "Odesa",
	searching: true,
};

export const WithError = Template.bind({});

WithError.args = {
	value: "Lorem Ipsum",
	errorMessage: "Search error",
};

export default {
	title: "CitySearch",
	component: CitySearch,
};
