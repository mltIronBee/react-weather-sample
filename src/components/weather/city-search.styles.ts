import { Theme } from "@material-ui/core";
import { makeStyles, createStyles } from "@material-ui/styles";

export default makeStyles((theme: Theme) =>
	createStyles({
		loadingIcon: {
			padding: theme.spacing() * 1.5,
		},
	}),
);
