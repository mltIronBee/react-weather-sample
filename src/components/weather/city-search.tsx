import React from "react";
import TextField from "@material-ui/core/TextField";
import InputAdornment from "@material-ui/core/InputAdornment";
import IconButton from "@material-ui/core/IconButton";
import SearchIcon from "@material-ui/icons/Search";
import CircularProgress from "@material-ui/core/CircularProgress";
import useStyles from "@components/weather/city-search.styles";

export interface ICitySearchProps {
	value: string;
	searching: boolean;
	errorMessage?: string;
	onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
	onSearch: (event: React.MouseEvent<HTMLButtonElement>) => void | Promise<void>;
}

export const CitySearch: React.FC<ICitySearchProps> = (props) => {
	const classes = useStyles();

	return (
		<TextField
			label="Search city"
			error={!!props.errorMessage}
			helperText={props.errorMessage}
			value={props.value}
			onChange={props.onChange}
			variant="outlined"
			InputProps={{
				endAdornment: (
					<InputAdornment position="end">
						{props.searching ? (
							<CircularProgress className={classes.loadingIcon} size={24} />
						) : (
							<IconButton onClick={props.onSearch}>
								<SearchIcon />
							</IconButton>
						)}
					</InputAdornment>
				),
			}}
		/>
	);
};
