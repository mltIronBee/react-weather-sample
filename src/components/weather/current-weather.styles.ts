import { makeStyles } from "@material-ui/core/styles";

export default makeStyles({
	sunriseSunsetContainer: {
		width: "100%",
		display: "flex",
		justifyContent: "space-evenly",
	},
	sunriseIcon: {
		color: "#EC6E4C",
	},
	sunsetIcon: {
		color: "#EC6E4C",
	},
	weatherContainer: {
		display: "flex",
		justifyContent: "center",
		alignItems: "center",
	},
	weatherDescription: {
		display: "inline-block",
		textTransform: "capitalize",
	},
	weatherIcon: {
		display: "inline-block",
	},
	windDirectionIcon: (windDeg: number) => ({
		transform: `rotate(${windDeg}deg)`,
	}),
});
