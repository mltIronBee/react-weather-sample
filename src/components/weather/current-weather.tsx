import React, { memo } from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import Divider from "@material-ui/core/Divider";
import useStyles from "@components/weather/current-weather.styles";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import DownIcon from "@material-ui/icons/VerticalAlignBottom";
import CloudIcon from "@material-ui/icons/Cloud";
import DropIcon from "@material-ui/icons/Opacity";
import SunIcon from "@material-ui/icons/WbSunny";
import SnowIcon from "@material-ui/icons/AcUnit";
import VisibilityIcon from "@material-ui/icons/Visibility";
import SpeedIcon from "@material-ui/icons/Speed";
import ArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import RainIcon from "@components/icons/rain";
import SunriseIcon from "@components/icons/sunrise";
import SunsetIcon from "@components/icons/sunset";

export interface ICurrentWeatherProps {
	date: string;
	sunrise: string;
	sunset: string;
	temperature: number;
	feelsLike: number;
	pressure: number;
	humidity: number;
	clouds: number;
	uvIndex: number;
	visibility: number;
	rain?: number;
	snow?: number;
	windSpeed: number;
	windDeg: number;
	weatherIcon: string;
	weatherDescription: string;
}

const formatTemperature = (temperature: number): string =>
	temperature > 0 ? `+${Math.round(temperature)}` : Math.round(temperature).toString(10);

const getReadableWindDirection = (windDeg: number): string => {
	if (windDeg > 337.5 || windDeg <= 22.5) {
		return "North";
	} else if (windDeg > 22.5 && windDeg <= 67.5) {
		return "North-East";
	} else if (windDeg > 67.5 && windDeg <= 112.5) {
		return "East";
	} else if (windDeg > 112.5 && windDeg <= 157.5) {
		return "South-East";
	} else if (windDeg > 157.5 && windDeg <= 202.5) {
		return "South";
	} else if (windDeg > 202.5 && windDeg <= 247.5) {
		return "South-West";
	} else if (windDeg > 247.5 && windDeg <= 292.5) {
		return "West";
	}

	return "North-West";
};

export const CurrentWeather: React.FC<ICurrentWeatherProps> = memo((props) => {
	const classes = useStyles(props.windDeg);

	return (
		<Grid container>
			<Grid item xs={12} md={5} lg={4} xl={3}>
				<Grid container direction="column" justify="center" alignItems="center">
					<Typography>{props.date}</Typography>
					<span className={classes.weatherContainer}>
						<img className={classes.weatherIcon} alt={props.weatherDescription} src={props.weatherIcon} />
						<Typography className={classes.weatherDescription} variant="h4">
							{props.weatherDescription}
						</Typography>
					</span>
					<Typography gutterBottom variant="h2">
						{formatTemperature(props.temperature)}&deg;
					</Typography>
					<Typography gutterBottom variant="h4">
						Feels like: {formatTemperature(props.feelsLike)}&deg;
					</Typography>
					<Typography gutterBottom component="span" className={classes.sunriseSunsetContainer}>
						<Typography align="center" variant="h5" display="inline">
							<SunriseIcon className={classes.sunriseIcon} /> Sunrise at {props.sunrise}
						</Typography>
						<Typography align="center" variant="h5" display="inline">
							<SunsetIcon className={classes.sunsetIcon} /> Sunset at {props.sunset}
						</Typography>
					</Typography>
				</Grid>
			</Grid>
			<Grid item xs={12} md={7} lg={8} xl={9}>
				<List>
					<ListItem>
						<ListItemIcon>
							<DownIcon />
						</ListItemIcon>
						<ListItemText primary={`${props.pressure} hPa`} secondary="Atmospheric pressure" />
					</ListItem>
					<Divider />
					<ListItem>
						<ListItemIcon>
							<DropIcon />
						</ListItemIcon>
						<ListItemText primary={`${props.humidity} %`} secondary="Humidity" />
					</ListItem>
					<Divider />
					<ListItem>
						<ListItemIcon>
							<CloudIcon />
						</ListItemIcon>
						<ListItemText primary={`${props.clouds} %`} secondary="Clouds" />
					</ListItem>
					<Divider />
					<ListItem>
						<ListItemIcon>
							<SunIcon />
						</ListItemIcon>
						<ListItemText primary={props.uvIndex} secondary="Midday UV Index" />
					</ListItem>
					<Divider />
					<ListItem>
						<ListItemIcon>
							<VisibilityIcon />
						</ListItemIcon>
						<ListItemText primary={`${props.visibility} m`} secondary="Visibility" />
					</ListItem>
					<Divider />
					{props.rain && (
						<React.Fragment>
							<ListItem>
								<ListItemIcon>
									<RainIcon />
								</ListItemIcon>
								<ListItemText primary={`${props.rain} mm`} secondary="Rain last 1 hour" />
							</ListItem>
							<Divider />
						</React.Fragment>
					)}
					{props.snow && (
						<React.Fragment>
							<ListItem>
								<ListItemIcon>
									<SnowIcon />
								</ListItemIcon>
								<ListItemText primary={`${props.snow} mm`} secondary="Snow last 1 hour" />
							</ListItem>
							<Divider />
						</React.Fragment>
					)}
					<ListItem>
						<ListItemIcon>
							<SpeedIcon />
						</ListItemIcon>
						<ListItemText primary={`${props.windSpeed} m/s`} secondary="Wind speed" />
					</ListItem>
					<Divider />
					<ListItem>
						<ListItemIcon>
							<ArrowUpwardIcon className={classes.windDirectionIcon} />
						</ListItemIcon>
						<ListItemText primary={getReadableWindDirection(props.windDeg)} secondary="Wind direction" />
					</ListItem>
				</List>
			</Grid>
		</Grid>
	);
});

CurrentWeather.displayName = "CurrentWeather";
