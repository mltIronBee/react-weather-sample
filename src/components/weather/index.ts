export * from "@components/weather/city-search";
export * from "@components/weather/forecast-graph";
export * from "@components/weather/tab-controls";
export * from "@components/weather/current-weather";
