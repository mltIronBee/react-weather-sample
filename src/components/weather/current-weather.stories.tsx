import React, { ComponentProps, useState } from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import Slider from "@material-ui/core/Slider";
import { DateTime } from "luxon";
import { Story } from "@storybook/react/types-6-0";
import { CurrentWeather, ICurrentWeatherProps } from "@components/weather";
import { getWeatherIcon } from "@utils/weather-api";

const WindDirectionWrapper: React.FC<Omit<ICurrentWeatherProps, "windDeg">> = (props) => {
	const [windDeg, setWindDeg] = useState(0);

	const handleWindDirectionChange = (event: React.ChangeEvent<unknown>, value: number | number[]): void => {
		setWindDeg(value as number);
	};

	return (
		<Grid container>
			<Typography gutterBottom>Wind direction</Typography>
			<Grid spacing={3} container>
				<Grid item>
					<Typography>0</Typography>
				</Grid>
				<Grid item xs>
					<Slider min={0} max={360} value={windDeg} onChange={handleWindDirectionChange} />
				</Grid>
				<Grid item>
					<Typography>360</Typography>
				</Grid>
			</Grid>
			<Grid container>
				<CurrentWeather windDeg={windDeg} {...props} />
			</Grid>
		</Grid>
	);
};

const Template: Story<ComponentProps<typeof WindDirectionWrapper>> = (args) => <WindDirectionWrapper {...args} />;

export const Default = Template.bind({});

Default.args = {
	date: DateTime.local().toLocaleString(DateTime.DATETIME_FULL),
	sunrise: "06:00",
	sunset: "18:00",
	temperature: 24,
	feelsLike: 20,
	pressure: 1000,
	humidity: 80,
	clouds: 0,
	uvIndex: 0,
	visibility: 10000,
	windSpeed: 5,
	weatherIcon: getWeatherIcon("01d"),
	weatherDescription: "Clear sky",
};

export const WithRain = Template.bind({});

WithRain.args = {
	date: DateTime.local().toLocaleString(DateTime.DATETIME_FULL),
	sunrise: "06:00",
	sunset: "18:00",
	temperature: 5,
	feelsLike: -3,
	pressure: 1000,
	humidity: 80,
	clouds: 0,
	uvIndex: 0,
	visibility: 10000,
	windSpeed: 5,
	weatherIcon: getWeatherIcon("09d"),
	weatherDescription: "shower rain",
	rain: 30,
};

export const WithSnow = Template.bind({});

WithSnow.args = {
	date: DateTime.local().toLocaleString(DateTime.DATETIME_FULL),
	sunrise: "06:00",
	sunset: "18:00",
	temperature: -14,
	feelsLike: -20,
	pressure: 1000,
	humidity: 80,
	clouds: 0,
	uvIndex: 0,
	visibility: 10000,
	windSpeed: 5,
	weatherIcon: getWeatherIcon("13d"),
	weatherDescription: "Snow",
	snow: 30,
};

export default {
	title: "Current weather component",
	component: CurrentWeather,
};
