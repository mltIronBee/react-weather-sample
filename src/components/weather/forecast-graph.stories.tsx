import React, { ComponentProps } from "react";
import { Story } from "@storybook/react/types-6-0";
import { ForecastGraph } from "@components/weather";
import { IForecastGraphProps } from "@components/weather/forecast-graph";

const sampleData: IForecastGraphProps["data"] = [
	{ dayOfWeek: 1, minTemperature: 5, maxTemperature: 13 },
	{ dayOfWeek: 2, minTemperature: 3, maxTemperature: 11 },
	{ dayOfWeek: 3, minTemperature: 4, maxTemperature: 9 },
	{ dayOfWeek: 4, minTemperature: 4, maxTemperature: 9 },
	{ dayOfWeek: 5, minTemperature: 4, maxTemperature: 9 },
	{ dayOfWeek: 6, minTemperature: 2, maxTemperature: 8 },
	{ dayOfWeek: 7, minTemperature: 1, maxTemperature: 8 },
];

const Template: Story<ComponentProps<typeof ForecastGraph>> = ({ data, ...args }) => (
	<ForecastGraph data={data || sampleData} {...args} />
);

export const Default = Template.bind({});

export const WithoutData = Template.bind({});

WithoutData.args = {
	data: [],
};

export const WithError = Template.bind({});

WithError.args = {
	hasError: true,
};

export const LoadingWithoutData = Template.bind({});

LoadingWithoutData.args = {
	data: [],
	loading: true,
};

export const LoadingWithData = Template.bind({});

LoadingWithData.args = {
	loading: true,
};

export const LoadingWithError = Template.bind({});

LoadingWithError.args = {
	loading: true,
	hasError: true,
};

export default {
	title: "Forecast chart component",
	component: ForecastGraph,
	decorators: [
		(Story: React.ComponentType): JSX.Element => (
			<div style={{ width: "50%", height: 400 }}>
				<Story />
			</div>
		),
	],
};
