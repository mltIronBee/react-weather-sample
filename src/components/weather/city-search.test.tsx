import React from "react";
import { render, screen } from "@utils/test-utils";
import { CitySearch } from "@components/weather";
import "@testing-library/jest-dom/extend-expect";

describe("City search input", () => {
	const voidCallback = (): void => undefined;

	describe("Rendering", () => {
		it("Should render correctly", () => {
			const testCity = "Odesa";
			const { container, getByDisplayValue } = render(
				<CitySearch
					geolocationAvailable
					searching={false}
					value={testCity}
					onGetLocation={voidCallback}
					onChange={voidCallback}
					onSearch={voidCallback}
				/>,
			);

			expect(getByDisplayValue(testCity)).toBeInTheDocument();
			expect(container.firstChild).toMatchSnapshot();
		});

		it("Should correctly render input during loading", () => {
			const { container } = render(
				<CitySearch
					geolocationAvailable
					onGetLocation={voidCallback}
					searching
					value=""
					onChange={voidCallback}
					onSearch={voidCallback}
				/>,
			);

			expect(container.firstChild).toMatchSnapshot();
		});

		it("Should correctly render error message and outline on error", () => {
			const testCity = "Lorem Ipsum";
			const testMessage = "Invalid city name";
			const { container, getByText, getByDisplayValue } = render(
				<CitySearch
					geolocationAvailable
					searching={false}
					errorMessage={testMessage}
					value={testCity}
					onGetLocation={voidCallback}
					onChange={voidCallback}
					onSearch={voidCallback}
				/>,
			);

			expect(getByDisplayValue(testCity)).toBeInTheDocument();
			expect(getByText(testMessage)).toBeInTheDocument();
			expect(container.firstChild).toMatchSnapshot();
		});
	});

	describe("Input controls managing", () => {
		it("Should disable search button, if user hasn't entered any text", async () => {
			render(
				<CitySearch
					value=""
					searching={false}
					geolocationAvailable
					onChange={voidCallback}
					onSearch={voidCallback}
					onGetLocation={voidCallback}
				/>,
			);

			const searchButton = screen.getByLabelText(/^search$/i);

			expect(searchButton).toBeDisabled();
		});

		it("Should enable search button, when user enters text", () => {
			render(
				<CitySearch
					value="test"
					searching={false}
					geolocationAvailable
					onChange={voidCallback}
					onSearch={voidCallback}
					onGetLocation={voidCallback}
				/>,
			);

			const searchButton = screen.getByLabelText(/^search$/i);

			expect(searchButton).not.toBeDisabled();
		});

		it("Should disable geolocation button, if geolocation is not available", () => {
			render(
				<CitySearch
					value=""
					searching={false}
					geolocationAvailable={false}
					onChange={voidCallback}
					onSearch={voidCallback}
					onGetLocation={voidCallback}
				/>,
			);

			const geolocationButton = screen.getByLabelText(/^geolocation is not available$/i);

			expect(geolocationButton).toBeDisabled();
		});

		it("Should not disable geolocation button, if geolocation is available", () => {
			render(
				<CitySearch
					geolocationAvailable
					value=""
					searching={false}
					onChange={voidCallback}
					onSearch={voidCallback}
					onGetLocation={voidCallback}
				/>,
			);

			const geolocationButton = screen.getByLabelText(/^use current location$/i);

			expect(geolocationButton).not.toBeDisabled();
		});
	});
});
