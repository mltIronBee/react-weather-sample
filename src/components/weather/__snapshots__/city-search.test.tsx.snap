// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City search input Rendering Should correctly render error message and outline on error 1`] = `
<form>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-error MuiInputLabel-error MuiFormLabel-filled"
      data-shrink="true"
      for="city-search"
      id="city-search-label"
    >
      Search city
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-error MuiOutlinedInput-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
    >
      <input
        aria-describedby="city-search-helper-text"
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
        id="city-search"
        placeholder="Enter city to search"
        type="text"
        value="Lorem Ipsum"
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <span
          class=""
          title="Use current location"
        >
          <button
            aria-label="use current location"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm8.94 3c-.46-4.17-3.77-7.48-7.94-7.94V1h-2v2.06C6.83 3.52 3.52 6.83 3.06 11H1v2h2.06c.46 4.17 3.77 7.48 7.94 7.94V23h2v-2.06c4.17-.46 7.48-3.77 7.94-7.94H23v-2h-2.06zM12 19c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          class=""
          title="Search"
        >
          <button
            aria-label="search"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </span>
      </div>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
        >
          <span>
            Search city
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-error MuiFormHelperText-filled"
      id="city-search-helper-text"
    >
      Invalid city name
    </p>
  </div>
</form>
`;

exports[`City search input Rendering Should correctly render input during loading 1`] = `
<form>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
      for="city-search"
      id="city-search-label"
    >
      Search city
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
        id="city-search"
        placeholder="Enter city to search"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <span
          class=""
          title="Use current location"
        >
          <button
            aria-label="use current location"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm8.94 3c-.46-4.17-3.77-7.48-7.94-7.94V1h-2v2.06C6.83 3.52 3.52 6.83 3.06 11H1v2h2.06c.46 4.17 3.77 7.48 7.94 7.94V23h2v-2.06c4.17-.46 7.48-3.77 7.94-7.94H23v-2h-2.06zM12 19c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </span>
        <div
          class="MuiCircularProgress-root makeStyles-loadingIcon MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 24px; height: 24px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled"
        >
          <span>
            Search city
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;

exports[`City search input Rendering Should render correctly 1`] = `
<form>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink="true"
      for="city-search"
      id="city-search-label"
    >
      Search city
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
        id="city-search"
        placeholder="Enter city to search"
        type="text"
        value="Odesa"
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <span
          class=""
          title="Use current location"
        >
          <button
            aria-label="use current location"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm8.94 3c-.46-4.17-3.77-7.48-7.94-7.94V1h-2v2.06C6.83 3.52 3.52 6.83 3.06 11H1v2h2.06c.46 4.17 3.77 7.48 7.94 7.94V23h2v-2.06c4.17-.46 7.48-3.77 7.94-7.94H23v-2h-2.06zM12 19c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          class=""
          title="Search"
        >
          <button
            aria-label="search"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </span>
      </div>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
        >
          <span>
            Search city
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</form>
`;
